fe/fe-core/src/test/java/com/starrocks/lake/qe/scheduler/DefaultSharedDataWorkerProviderTest.java

```diff
@@ -299,7 +299,26 @@ public void testSelectNextWorker() throws UserException {
         { // test no available worker to select
             WorkerProvider workerProvider =
                     new DefaultSharedDataWorkerProvider(ImmutableMap.copyOf(id2AllNodes), ImmutableMap.of());
-            Assert.assertThrows(NonRecoverableException.class, workerProvider::selectNextWorker);
+
+            Exception e = Assert.assertThrows(NonRecoverableException.class, workerProvider::selectNextWorker);
+            Assert.assertEquals(
+                    "Compute node not found. Check if any compute node is down. nodeId: -1 " +
+                            "compute node: [host#1 alive: true, available: false, inBlacklist: false] " +
```

Comment:
this might be misleading. 

```
new DefaultSharedDataWorkerProvider(ImmutableMap.copyOf(id2AllNodes), ImmutableMap.of());
```
This is created just for simulating no available workers. But in real code path, the provider is created by the DefaultSharedDataWorkerProvider.Factory, which the `availableComputeNodes` is a filtering result of `idToComputeNode`, and the filter rule is either the node is dead or the node is in Blacklist.

Might need to revise this case to make it more realistic.